src/network/networkd-routing-policy-rule.c

```diff
@@ -1622,6 +1643,54 @@ int config_parse_routing_policy_rule_suppress_prefixlen(
         return 0;
 }
 
+int config_parse_routing_policy_rule_suppress_ifgroup(
+                const char *unit,
+                const char *filename,
+                unsigned line,
+                const char *section,
+                unsigned section_line,
+                const char *lvalue,
+                int ltype,
+                const char *rvalue,
+                void *data,
+                void *userdata) {
+
+        _cleanup_(routing_policy_rule_free_or_set_invalidp) RoutingPolicyRule *n = NULL;
+        Network *network = userdata;
+        int r;
+
+        assert(filename);
+        assert(section);
+        assert(lvalue);
+        assert(rvalue);
+        assert(data);
+
+        r = routing_policy_rule_new_static(network, filename, section_line, &n);
+        if (r < 0)
+                return log_oom();
+
+        if (isempty(rvalue)) {
+                n->suppress_ifgroup = -1;
+                return 0;
+        }
+
+        r = safe_atoi32(rvalue, &n->suppress_ifgroup);
+        if (r < 0) {
+                n->suppress_ifgroup = -1;
+                log_syntax(unit, LOG_WARNING, filename, line, r,
+                           "Failed to parse SuppressIfGroup=, ignoring assignment: %s", rvalue);
+                return 0;
+        }
+        if (n->suppress_ifgroup < 0) {
+                log_syntax(unit, LOG_WARNING, filename, line, r,
+                           "Failed to parse SuppressIfGroup=, value should be in range: [0, 2147483647]");
```

Comment:
Hmm, this message is misleading, since it *was* parsed successfully. Just say "Invalid SuppressIfGroup= value %"PRIi32", must be in the range 0…2147483647". 

Also use a temporary variable and avoid setting the output variable on failure.